File: app/src/main/java/pl/piotrskiba/angularowo/models/Rank.java
package pl.piotrskiba.angularowo.models;

public class Rank {

    private final String name;
    private final boolean staff;
    private final String colorCode;
    private final String chatColorCode;

    public Rank(String name, boolean staff, String colorCode, String chatColorCode){
        this.name = name;
        this.staff = staff;
        this.colorCode = colorCode;
        this.chatColorCode = chatColorCode;
    }

    public String getName() {
        return name;
    }

    public boolean isStaff() {
        return staff;
    }

    public String getColorCode() {
        return colorCode;
    }

    public String getChatColorCode() {
        return chatColorCode;
    }
}

File: domain/src/main/java/pl/piotrskiba/angularowo/domain/rank/model/Rank.kt
package pl.piotrskiba.angularowo.domain.rank.model

import java.io.Serializable

sealed class Rank(
    val name: String,
    val staff: Boolean,
    val colorCode: String,
    val chatColorCode: String
) : Serializable {
    data class UnknownRank(val _name: String) : Rank(_name, false, "7", "7")
    data class CustomRank(
        val _name: String,
        val _staff: Boolean,
        val _colorCode: String,
        val _chatColorCode: String
    ) : Rank(_name, _staff, _colorCode, _chatColorCode)
}
